VideoLocker/src/main/java/org/edx/mobile/http/HttpManager.java

```diff
@@ -301,17 +307,58 @@ private HttpParams getHttpParameters() {
 
         HttpGet get = new HttpGet(url);
 
+        // set request headers
+        if (headers != null) {
+            for (String key : headers.keySet()) {
+                get.addHeader(key, headers.getString(key));
+            }
+        }
+
         HttpResponse response = client.execute(get);
         org.apache.http.Header header=null;
         if(response.containsHeader("Set-Cookie")){
             header = response.getFirstHeader("Set-Cookie");
         }
-        
         client.getConnectionManager().shutdown();
 
         return header;
     }
 
+    public org.apache.http.Header getResponseHeader(String url)
+        throws ParseException, ClientProtocolException, IOException {
+        return getResponseHeader(url, null);
+    }
+
+    public List<Cookie> getCookies(String url, Bundle headers)
```

Comment:
@aleffert  yeah.. this method is not called for this story. i add it as we dont have any support for using cookie in http request. and original code to get cookie is buggy. (getRequestHeader() )